setup.cfg

```diff
@@ -31,6 +30,8 @@ classifiers =
 
 [options]
 python_requires = >=3.6
+setup_requires =
+    setuptools_scm>=30.3.0
```

Comment:
whats the difference between this one and above

Reply:
What do you mean by above?

Reply:
in the pyproject.toml

Reply:
and why is the version not pinned?

Reply:
Yeah emm.. good point. You can read up on the documentation [here](https://pypi.org/project/setuptools-scm/). I'm not 100% sure, I just added this statement here as well, assuming `pip` or `setup.py` uses `setup.cfg`, not `pyproject` (while `setuptools-scm` merely uses `pyproject.toml` as a config).

Reply:
> and why is the version not pinned?

Oh but this I certainly wouldn't do... That would mean that you require the install system to have one specific version. Almost like requiring one specific version of python.

Reply:
Hmm, but we are not fixing Python just all the dependencies right? I just want to avoid any troubles with incompatibilities

Reply:
I think we only want to fix dependencies that come during the install stage, that is, the ones listed under `install_requires`.